compiler/rustc_error_messages/locales/en-US/hir_analysis.ftl

```diff
@@ -137,3 +137,5 @@ hir_analysis_expected_used_symbol = expected `used`, `used(compiler)` or `used(l
 hir_analysis_missing_parentheses_in_range = can't call method `{$method_name}` on type `{$ty_str}`
 
 hir_analysis_add_missing_parentheses_in_range = you must surround the range in parentheses to call its `{$func_name}` function
+
+hir_analysis_self_in_impl_self = `Self` is not valid at this location
```

Comment:
```suggestion
hir_analysis_self_in_impl_self = `Self` is not valid in the `impl` type
```

Reply:
Can we add a span label pointing at the `Self` on its own with the text "not valid at this location"?

Reply:
I think that we cant since `#[note] spans: MultiSpan` is an error, I think the errors are also relatively clear as to what the problematic location is even without that span being there so its not a _huge_ loss. :thinking: